### ISSUE TITLE
TypedVal float_val only allows float32

### ISSUE THREAD

**Issue by aaronbee** [openconfig/gnmi#54]
`TypedValue` in gnmi.proto has a `float_val` member whose type is `float`. This translates to a 32-bit floating point in generated code, eg. a `float32` in Go. The intention based on the gNMI Specification seems to have been for this to be a `double` which translates to a 64-bit floating point, eg. `float64` in Go.

[From the gNMI Specification:
](https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md#223-node-values)
> `float` in the `float_val` field, used to store floating-point values (i.e., `float32`, `float64`).

Unfortunately, this type can't be changed to `double` without breaking backwards compatibility. A new `double` field could be introduced in `TypedValue`, and possibly the `float` field could be deprecated. Or, it could be advised to encode `float64` values as JSON.

**Comment by gcsl** [comment]
What variable needs either the precision or range of a double and cannot be
accommodated by a 32 but float? All floating point values in the OpenConfig
schema come from measurements that are derived from A/D converters, are
they not?

**Comment by aaronbee** [comment]
I was more pointing out that the gNMI specification is out of sync with gnmi.proto: the spec suggests there would be a place for float64 values in TypeVal, but that's not the case. That's fine if that is an intentional decision.

**Comment by aaronbee** [comment]
Bumping this issue. Arista uses gNMI for more things than just OpenConfig models. A lot of our internal data uses float64. Any chance this could be added to a TypedValue?

**Comment by gcsl** [comment]
Whether OpenConfig or not, before considering whether to support 64bit
floating point values in gNMI, is there a compelling reason to do so for
need of either the extended precision or magnitude over float32?  The
question isn't whether float64 is used somewhere else but whether where it
is used, was it chosen out of necessity?  The problem arises in that once
there are multiple floating types in TypedValue, every future
implementation must decide between them and likely the choice will be made
arbitrarily or naively (64 must obviously be better than 32).  In the
integer case, there are quite valid reasons for needing a 64bit value as
certain counters can exceed the 32bit limit in a very short time.  On the
integer side we chose only 64bit values in gNMI for all integers to capture
all use cases.  In the floating point case, we chose only the 32bit option
to capture all use cases as not a single case was identified as needing
64bit types.
